language-server/src/server.js

```diff
@@ -133,7 +133,7 @@ async function getDocumentSettings(resource) {
     documentSettings.set(resource, result);
   }
 
-  return result;
+  return result ?? {};
```

Comment:
@jdesrosiers Now we're returning an empty object when settings is `null`. This is done to ensure that in `neovim` the server doesn't crashes when settings is `null`.
Tested in both VSCode and Neovim.

Reply:
Although this works, wouldn't it be cleaner to never put a `null` in the `documentSettings` Map in the first place?

Reply:
yes, so maybe storing `result` in `documentSettings` only when it is not null should be the solution.

Reply:
Yes that'd be better. Solved by @diyaayay in this commit 93372c1